### DISCUSSION TITLE
Altering Foreign Key doesn't create Constraint

### DISCUSSION THREAD

**Discussion by guilhermemiua** [adonisjs/core#1461] [Help]
Hi, I'm facing a problem to alter some Foreign Keys in my Project.

I drop the Foreign Key then I alter the column. When I rollback the migrations appears an error saying that "constraint does not exist". Here it is my Migration File.

`'use strict'

/** @type {import('@adonisjs/lucid/src/Schema')} */
const Schema = use('Schema')

class AlterDocSchema extends Schema {
  up () {
    this.alter('doc', async (table) => {
      await table.dropForeign('status_group_id')
      
      await table
        .integer('status_group_id')
        .unsigned()
        .references('id')
        .inTable('status_groups')
        .onUpdate('CASCADE')
        .notNullable()
        .alter()
    })
  }

  down () {
    this.alter('doc', async (table) => {
      await table.dropForeign('status_group_id')
     
      await table
        .integer('status_group_id')
        .unsigned()
        .references('id')
        .inTable('status_groups')
        .onUpdate('CASCADE')
        .onDelete('CASCADE')
        .notNullable()
        .alter()
    })
  }
}

module.exports = AlterDocSchema
`

What's the correct way to alter a foreign key column?

**Comment by thetutlage** [comment]
Please format the code blocks. https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown

**Reply by guilhermemiua** [reply]
Hi @thetutlage ! Ok!

```
'use strict'

/** @type {import('@adonisjs/lucid/src/Schema')} */
const Schema = use('Schema')

class AlterDocSchema extends Schema {
  up () {
    this.alter('doc', async (table) => {
      await table.dropForeign('status_group_id')
      
      await table
        .integer('status_group_id')
        .unsigned()
        .references('id')
        .inTable('status_groups')
        .onUpdate('CASCADE')
        .notNullable()
        .alter()
    })
  }

  down () {
    this.alter('doc', async (table) => {
      await table.dropForeign('status_group_id')
     
      await table
        .integer('status_group_id')
        .unsigned()
        .references('id')
        .inTable('status_groups')
        .onUpdate('CASCADE')
        .onDelete('CASCADE')
        .notNullable()
        .alter()
    })
  }
}

module.exports = AlterDocSchema

```

**Reply by guilhermemiua** [reply]
I have a foreign key in "doc" table which I want to remove the CASCADE onDelete. I'm trying to alter this table, but when I delete the FK Constraint and alter the column, the constraint is not recreated. I'd like to know what is the correct way to alter a FK.

**Reply by thetutlage** [reply]
Where exactly in the docs you find that you have to use `await` with `table.dropForeign` and `table.integer`?
